sdk/eventgrid/azure-resourcemanager-eventgrid/src/main/samples/java/com/azure/resourcemanager/eventgrid/EventGridPublishAndConsumeExample.java

```diff
@@ -0,0 +1,234 @@
+                .endpoint(eventGridTopic.endpoint())
+                .credential(new AzureKeyCredential(eventGridClientConnectionKey))
+                .buildEventGridEventPublisherClient();
+
+            System.out.println("Done creating event grid publisher client.");
+
+            // 7. Create an event hub consumer client
+            String connectionString = eventHub.listAuthorizationRules().stream().findFirst().get().getKeys().primaryConnectionString();
+            System.out.format("Event hub connection string: %s%n", connectionString);
+
+            EventHubConsumerAsyncClient consumer = new EventHubClientBuilder()
+                .connectionString(connectionString)
+                .consumerGroup(EventHubClientBuilder.DEFAULT_CONSUMER_GROUP_NAME)
+                .buildAsyncConsumerClient();
+
+            System.out.println("Done creating event hub consumer client.");
+
+            // 8. Subscribe to coming events from event grid using `EventHubConsumerAsyncClient`.
+            // 8.1 Get the first partition and block on it
+            String firstPartition = consumer.getPartitionIds().blockFirst(OPERATION_TIMEOUT);
+
+            // This shouldn't happen, but if we are unable to get the partitions within the timeout period.
+            if (firstPartition == null) {
+                firstPartition = "0";
+            }
+
+            // 8.2 Subscribe to the coming events from event grid
+            // We use CountDownLatch to wait for the coming events
+            CountDownLatch countDownLatch = new CountDownLatch(NUMBER_OF_EVENTS);
+            consumer.receiveFromPartition(firstPartition, EventPosition.latest())
+                .subscribe(partitionEvent -> {
+                        EventData eventData = partitionEvent.getData();
+                        String contents = new String(eventData.getBody(), UTF_8);
+                        countDownLatch.countDown();
+
+                        System.out.printf("Event received. Event sequence number number: %s. Contents: %s%n",
+                            eventData.getSequenceNumber(), contents);
+                    },
+                    error -> {
+                        System.err.println("Error occurred while consuming events: " + error);
+
+                        // Count down until 0, so the main thread does not keep waiting for events.
+                        while (countDownLatch.getCount() > 0) {
+                            countDownLatch.countDown();
+                        }
+                    }, () -> {
+                        System.out.println("Finished reading events.");
+                    });
+
+            // 9. Publish custom events to the EventGrid.
+            // We create events to send to the service and block until the send has completed.
+            Flux.range(0, NUMBER_OF_EVENTS)
+                .doOnNext(number -> {
+                    String body = String.format("Hello world! Number: %s", number);
```

Comment:
Hello world! ???

Reply:
sample in azure-messaging-eventhubs uses 'Hello World' as event body as well, so I keep the name. We can change it to `String body = String.format("Custom event number: %s", number);`